### ISSUE TITLE
[BUG] Table doesn't keep nlt features, if you edit the source markdown

### ISSUE THREAD

**Issue by thomas-esau** [trey-wallis/obsidian-notion-like-tables#200]
**Describe the bug**
When you edit the source markdown, the table will not update. Maybe linked to #41 and the mentioned tickets within this ticket.

**To Reproduce**
1. Delete the data.json file of nlt
2. Open up Obsidian
3. Create a markdown table
4. Render the table with preview mode; a nlt table gets rendered
5. Switch back to edit mode. Add a row.
6. Render the table with preview mode again; a normal table without nlt features gets rendered.

**Expected behavior**
If a table gets altered in edit mode, it should keep the nlt features.

**Screenshots**
![Animation](https://user-images.githubusercontent.com/46598525/176137717-d86cde2f-0600-4c73-a33b-8751f25962d8.gif)

**Debug Info**
SYSTEM INFO:
	Obsidian version: v0.15.3
	Installer version: v0.14.15
	Operating system: Windows 10 Education 10.0.19043
	Login status: logged in
	Catalyst license: insider
	Insider build toggle: on
	Live preview: on
	Legacy editor: off
	Base theme: dark
	Community theme: none
	Snippets enabled: 0
	Safe mode: off
	Plugins installed: 2
	Plugins enabled: 2
		1: Advanced Tables
		2: Notion-Like Tables

RECOMMENDATIONS:
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

**Using Mobile App?**
No

**Comment by trey-wallis** [comment]
Hi @thomas-esau. Thank you for the bug report. Yes, it seems like there is an error that is occurring. (The NLT table will not render when there is a plugin error). Does this only happen when you add a new row via markdown and then switch back to reading mode? Is it possible for you to send me a screenshot of your console?

`OPTION-CMD-I` on Mac
`CTRL-ALT-I` on Windows (I believe)

**Comment by hustrjh** [comment]
**Error info**
```
Uncaught (in promise) TypeError: Cannot destructure property 'creationTime' of 'oldData.rows[i]' as it is undefined.
    at eval (plugin:notion-like-tables:59681:15)
    at Array.forEach (<anonymous>)
    at updateAppDataFromSavedState (plugin:notion-like-tables:59679:16)
    at eval (plugin:notion-like-tables:59739:25)
    at Generator.next (<anonymous>)
    at eval (plugin:notion-like-tables:66:61)
    at new Promise (<anonymous>)
    at __async (plugin:notion-like-tables:50:10)
    at loadAppData (plugin:notion-like-tables:59703:57)
    at NLTTable.eval (plugin:notion-like-tables:59784:42)
```
---
Hi, @trey-wallis. Actually I met the same problem. I don't know if I get the error information you want. This is what I found in the console pane. I also found that if you deleted rows you added in souce markdown, the NLT table will render normally and function well. I hope the message would be helpful.
P.S. I used the **`Advanced tables`** plugin as well.

**Comment by trey-wallis** [comment]
@hustrjh Hi hustrijh. Ah, I see what the issue is now. I will look into pushing out a fix for that today. Which version of the table are you using?

**Comment by hustrjh** [comment]
@trey-wallis Hi trey-wallis. This is the source of wrong table in my data.json. I also tested that the NLT table would render incorrectly when I changed any cell. In this case the NLT table couldn't render normally again even when I undone the operation. There's time difference. So take your time. Thank you for your efforts. :smile: 

```
"Tests/test.md": {
      "0": {
        "data": {
          "headers": [
            {
              "id": "2770f284-e402-46d8-87ac-1bbf3f3d4c28",
              "initialIndex": 0,
              "content": "title",
              "sortName": "default",
              "width": "100px",
              "shouldWrapOverflow": true,
              "useAutoWidth": false,
              "type": "text"
            },
            {
              "id": "7e5690b1-d682-49db-adb3-27132b1ba881",
              "initialIndex": 1,
              "content": "text",
              "sortName": "default",
              "width": "100px",
              "shouldWrapOverflow": true,
              "useAutoWidth": false,
              "type": "text"
            },
            {
              "id": "b89413b8-8d21-4d10-97aa-6c20fc6a782d",
              "initialIndex": 2,
              "content": "blabla",
              "sortName": "default",
              "width": "100px",
              "shouldWrapOverflow": true,
              "useAutoWidth": false,
              "type": "tag"
            }
          ],
          "rows": [
            {
              "id": "b9bdb60a-53ec-4012-8b6c-779f58948fd7",
              "initialIndex": 0,
              "creationTime": 1656559168343
            }
          ],
          "cells": [
            {
              "id": "b560c110-5d30-4665-bc27-25f1d7f75238",
              "rowId": "b9bdb60a-53ec-4012-8b6c-779f58948fd7",
              "headerId": "2770f284-e402-46d8-87ac-1bbf3f3d4c28",
              "type": "text",
              "text": "item1"
            },
            {
              "id": "36651bdb-d3c8-4abb-b1c5-0bc680a9e777",
              "rowId": "b9bdb60a-53ec-4012-8b6c-779f58948fd7",
              "headerId": "7e5690b1-d682-49db-adb3-27132b1ba881",
              "type": "text",
              "text": "text1"
            },
            {
              "id": "23485c2a-dbd3-435c-ad3e-403c1128d652",
              "rowId": "b9bdb60a-53ec-4012-8b6c-779f58948fd7",
              "headerId": "b89413b8-8d21-4d10-97aa-6c20fc6a782d",
              "type": "tag"
            }
          ],
          "tags": [
            {
              "id": "03a02bb9-6074-4f65-9325-12e3e52e1026",
              "headerId": "b89413b8-8d21-4d10-97aa-6c20fc6a782d",
              "content": "111",
              "color": "pink",
              "selected": [
                "23485c2a-dbd3-435c-ad3e-403c1128d652"
              ]
            },
            {
              "id": "97cff231-7de2-4168-8404-e8fb401c1a52",
              "headerId": "b89413b8-8d21-4d10-97aa-6c20fc6a782d",
              "content": "gr",
              "color": "yellow",
              "selected": [
                "8b7a4dc1-ec95-47d7-94fd-5661e045f6ca"
              ]
            }
          ]
        },
        "viewType": "reading",
        "shouldUpdate": true,
        "tableCacheVersion": 410
      }
    }
```

**Comment by trey-wallis** [comment]
@thomas-esau @hustrjh Fixed in 4.1.1. Thanks for all the debugging info!
